### DISCUSSION TITLE
Future plans for pywinauto?

### DISCUSSION THREAD

**Discussion by rp2700** [pywinauto/pywinauto#1311] [General]
Hello,

This is a really good tool and I am able to automate different types of GUI element for AUT.
But I am concerned about the future releases. Are there any improvements / bug fixes / releases pipe lined? As I can see the last release was in 2019 so are there any migration plans or upgrades for this tool.

Requesting to please comment on this.

**Comment by junkmd** [comment]
These are the pinned issues;

- https://github.com/pywinauto/pywinauto/issues/328
- https://github.com/pywinauto/pywinauto/issues/335
- https://github.com/pywinauto/pywinauto/issues/332

And several PRs related to these have been submitted (although there have been no recent activities).

I am also a maintainer of `comtypes`, one of the dependencies of this package, and have a very strong interest in this package as well.
I have submitted the following issue.
https://github.com/pywinauto/pywinauto/issues/1265

**Reply by junkmd** [reply]
I think macOS support may be quite difficult.

**Reply by rp2700** [reply]
We are actually looking out for a good automation tool from long term perspective which will have active community and software updates. Pywinauto is really good tool and we are checking from support/community perspective as we want to migrate our large tests from coded ui to some other tool. Some guidance / comments would be appreciated. Thanks

**Reply by junkmd** [reply]
@rp2700 

I think you and I are quite similar in our thoughts on "Pywinauto is a really good tool".

This package implements many wrapper classes that make `win32functions` and `UIAutomationCore.dll` easy to use, and has an excellent set of very easy-to-use features.
When my project was launched, `pywinauto` was released with version-updates at least once every six months to a year.
For those reasons, this wonderful package was adopted for my project.

However, recently the releases have stopped while consultation on use cases continues to be posted on disscussions and Issues.
And I feel that @vasily-v-ryabov, the main maintainer of this package, is lacking in resources.

I would like to help something about this package.
I have recently been commenting on discussions and issues to motivate and develop the community for this package, even if only in a small way.

**Reply by junkmd** [reply]
This package still supports Python2.7.

I would share what I know about [`comtypes`](https://pypi.org/project/comtypes/) and [`pywin32`](https://pypi.org/project/pywin32/) on which this package depends.

Compared to the number of [stars](https://github.com/enthought/comtypes/stargazers) and [watchers](https://github.com/enthought/comtypes/watchers) in the GitHub repository for `comtypes`, the number of projects that depend on it should be enormous, as indicated by [the libraries.io](https://libraries.io/pypi/comtypes).
And my project depends strongly on it.
However, the `comtypes` codebase was pretty legacy style, so over the past year I have focused on refactoring (and adding a few features) to that codebase, and have become a repository collaborator of `comtypes`.
Several new backward-compatible versions of `comtypes` have been released. And I am planning to make some non-backward-compatible changes to `comtypes`, such as dropping Python2 support and adding runtime code inline type annotations.

`pywin32` is no longer supporting Python2(The CI testing is passed because a suitable version for Python 2.7 is installed by the `wheel`).

**Reply by rp2700** [reply]
@junkmd - Thank you for commenting!
@vasily-v-ryabov - Can you also please comment about this as we are looking from a long term solution? Are there any releases planned for the future? Any enhancements / fixes going on?

**Reply by vasily-v-ryabov** [reply]
Hi @rp2700 and thanks for your interest! Yes, I have plans to get back to maintaining pywinauto. Unfortunately internal complexity (despite external easiness) doesn't attract a lot of contributors. Currently I still have no much resources to finish macOS, Linux and WPF dll injection backends with some re-factoring on Windows in the `atspi` branch. I think this autumn or winter will be better time to revive my contributions. This is a hobby for me, and current personal activities a little bit displaced pywinauto, but these activities are expected to mostly finish till October.

**Reply by junkmd** [reply]
Hello @vasily-v-ryabov !

I think I could contribute a bit to the Windows refactoring.
Once my summer vacation is over, I plan to start working on some refactoring.

**Reply by vasily-v-ryabov** [reply]
@junkmd the only thing I would prefer is making functional refactoring first, only then we can think about adding typing. 😄

**Comment by cassini1010** [comment]
Happy to know that pywinauto is going to get updates in future 😊. I'm also using pywinauto for UI automation project in my organization.
